<a href="https://colab.research.google.com/github/BRLTS/SMKDEV/blob/main/TanyaJawabAI_VitoRozaan.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Tanya Jawab AI Menggunakan Gemini

# By Vito Rozaan Alandeta | SSEC Kelas #8

# Isi Notebook:
✅ Instalasi library google-generativeai

✅ Konfigurasi API Key Gemini

✅ Fungsi menampilkan riwayat

✅ Fungsi chat tanya-jawab

✅ Menyimpan jawaban ke riwayat.txt

✅ Fitur unduh riwayat

In [1]:
# 1. Install Library Gemini Terbaru
!pip install -q google-generativeai

# 2. Import dan Setup API Key
import google.generativeai as genai

# Ganti ini dengan API key dari https://aistudio.google.com/app/apikey
genai.configure(api_key="__________")

# 3. Load Model Gemini
try:
    model = genai.GenerativeModel("gemini-1.5-flash")  # atau "gemini-1.5-pro"
    print("Model berhasil dimuat.")
except Exception as e:
    print(f"Gagal memuat model: {e}")


Model berhasil dimuat.


In [4]:
# Menampilkan Riwayat Q&A
def show_history(n=3):
    try:
        with open("riwayat.txt", "r", encoding="utf-8") as f:
            lines = f.readlines()
            last_entries = lines[-n*3:]
            print("\n--- Last History ---")
            print("".join(last_entries))
    except FileNotFoundError:
        print("No history found.\n")

# Fitur Chat AI
def chat_ai():
    show_history()

    while True:
        question = input("Masukkan pertanyaan (atau 'keluar' untuk berhenti): ")
        if question.lower() == "keluar":
            print("Terima kasih sudah bertanya.")
            break

        try:
            response = model.generate_content(question)
            answer = response.text
        except Exception as e:
            answer = f"[Terjadi Error: {e}]"

        print("\nJawaban AI:\n", answer)

        # Simpan Riwayat
        with open("riwayat.txt", "a", encoding="utf-8") as f:
            f.write(f"Q: {question}\nA: {answer}\n\n")

# Launch Chat
chat_ai()

Belum ada riwayat.

Masukkan pertanyaan (atau 'keluar' untuk berhenti): apa itu umamusume

Jawaban AI:
 Umamusume Pretty Derby (ウマ娘 プリティーダービー) adalah sebuah franchise multimedia Jepang yang menampilkan karakter antropomorfik dari kuda pacu terkenal.  Franchise ini mencakup berbagai media, termasuk:

* **Anime:**  Serial anime utama adalah titik awal yang paling dikenal banyak orang, menceritakan kisah gadis-gadis yang merupakan reinkarnasi dari kuda pacu legendaris, yang berjuang untuk mencapai puncak dunia pacuan kuda.

* **Game mobile:** Sebuah game mobile gacha yang sangat populer, di mana pemain mengumpulkan dan melatih Umamusume mereka sendiri, berlomba melawan pemain lain, dan meningkatkan kemampuan mereka.  Ini adalah inti dari franchise dan seringkali menjadi pintu masuk utama bagi penggemar baru.

* **Manga:** Beberapa manga telah diterbitkan, yang memperluas cerita dan karakter dari franchise.

* **Game arcade:** Game arcade juga tersedia.

Pada dasarnya, Umamusume adalah cam

ERROR:tornado.access:503 POST /v1beta/models/gemini-1.5-flash:generateContent?%24alt=json%3Benum-encoding%3Dint (127.0.0.1) 5936.60ms
ERROR:tornado.access:503 POST /v1beta/models/gemini-1.5-flash:generateContent?%24alt=json%3Benum-encoding%3Dint (127.0.0.1) 3890.66ms



Jawaban AI:
 Tidak ada satu karakter utama tunggal dalam seri *Uma Musume Pretty Derby*.  Cerita berfokus pada *ensemble cast*,  sekelompok gadis-gadis yang merupakan inkarnasi dari kuda pacu terkenal.  Meskipun beberapa karakter memiliki peran lebih besar dan lebih banyak sorotan daripada yang lain,  tidak ada karakter yang secara konsisten menjadi tokoh pusat di semua cerita.

Namun, beberapa karakter yang sering dianggap sebagai tokoh utama atau karakter sentral, antara lain:

* **Special Week:**  Sering menjadi pusat cerita di musim pertama anime.
* **T M Opera O:** Memiliki peran penting dan sorotan yang signifikan di sepanjang seri.
* **Silent Bloom:** Memiliki peran krusial di musim kedua dan sering menjadi fokus cerita.

Pada akhirnya, siapa yang dianggap sebagai "karakter utama"  sangat subjektif dan bergantung pada preferensi masing-masing penonton.  Kekuatan *Uma Musume* terletak pada *ensemble cast*-nya dan hubungan antar karakternya.

Masukkan pertanyaan (atau 'keluar' un

ERROR:tornado.access:503 POST /v1beta/models/gemini-1.5-flash:generateContent?%24alt=json%3Benum-encoding%3Dint (127.0.0.1) 582.31ms
ERROR:tornado.access:503 POST /v1beta/models/gemini-1.5-flash:generateContent?%24alt=json%3Benum-encoding%3Dint (127.0.0.1) 581.99ms
ERROR:tornado.access:503 POST /v1beta/models/gemini-1.5-flash:generateContent?%24alt=json%3Benum-encoding%3Dint (127.0.0.1) 2122.19ms
ERROR:tornado.access:503 POST /v1beta/models/gemini-1.5-flash:generateContent?%24alt=json%3Benum-encoding%3Dint (127.0.0.1) 5683.25ms
ERROR:tornado.access:503 POST /v1beta/models/gemini-1.5-flash:generateContent?%24alt=json%3Benum-encoding%3Dint (127.0.0.1) 607.16ms
ERROR:tornado.access:503 POST /v1beta/models/gemini-1.5-flash:generateContent?%24alt=json%3Benum-encoding%3Dint (127.0.0.1) 683.39ms
ERROR:tornado.access:503 POST /v1beta/models/gemini-1.5-flash:generateContent?%24alt=json%3Benum-encoding%3Dint (127.0.0.1) 582.09ms
ERROR:tornado.access:503 POST /v1beta/models/gemini-1.5-flash:gener


Jawaban AI:
 Maruzensky adalah nama kuda pacu legendaris Jepang yang lahir pada tahun 1990. Ia terkenal karena prestasi gemilangnya di berbagai perlombaan kelas atas, terutama di Jepang.  Meskipun detail spesifik prestasi dan silsilahnya memerlukan penelusuran lebih lanjut di basis data balap kuda,  nama "Maruzensky" terkait erat dengan kuda-kuda pacu berkualitas tinggi di Jepang.  Nama ini mungkin juga merujuk pada nama pemilik, peternak, atau pelatih yang terkait dengan kuda-kuda terkenal tersebut.  Untuk informasi yang lebih spesifik tentang kuda pacu tertentu yang bernama Maruzensky, Anda perlu menyebutkan nama lengkapnya (misalnya, Maruzensky Arch).

Masukkan pertanyaan (atau 'keluar' untuk berhenti): keluar
Terima kasih sudah bertanya.


In [5]:
from google.colab import files
files.download("riwayat.txt")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# Refleksi Pribadi

> Apa tantangan utama dalam membangun aplikasi ini?

Tantangan terbesar dalam pembuatan program ini adalah memilih API AI yang tepat. Maksud dari "tepat" di sini adalah API yang bisa digunakan secara gratis. Awalnya, saya mencoba menggunakan OpenAI, namun ternyata API tersebut memerlukan akun berbayar (pro) agar bisa digunakan. Selain itu, saya juga mengalami berbagai error dan peringatan saat menjalankan kode, seperti Error Code 503 yang terus muncul.

> Apa pelajaran paling penting tentang penggunaan API?

Hal paling penting yang saya pelajari adalah bahwa API Key merupakan semacam kata sandi rahasia yang memberikan akses ke layanan AI. Jika API Key tidak dijaga dengan baik, orang lain bisa menyalahgunakannya, yang dapat menyebabkan tagihan di akun kita membengkak akibat penggunaan yang tidak sah atau berlebihan.


> Bagian mana dari aplikasi yang kamu banggakan?

Saya bangga dengan keseluruhan program ini. Ini adalah pertama kalinya saya berhasil membuat sebuah aplikasi chat sederhana yang terhubung dengan AI generatif. Saya merasa senang dan puas karena bisa membangun program seperti ini dari awal.

> Jika memiliki lebih banyak waktu, apa yang ingin kamu kembangkan lebih lanjut?

Jika saya memiliki lebih banyak waktu, saya ingin mengembangkan aplikasi ini menjadi semacam chat AI dengan karakter tertentu yang bisa diatur oleh user. Atau, saya juga ingin mengembangkannya menjadi aplikasi asisten pendukung (support app) yang dapat membantu user secara interaktif.